README.md

```diff
@@ -0,0 +1,87 @@
+
+```
+1. Send the search terms to the server that will fetch our results.
+3. Display the results.
+```
+
+This is simple indeed, but the fetching of results from the server -- just as the gathering of ingredients in our peanut butter sandwich example -- will take an indeterminate amount of time. So how do we handle this scenario?
+
+The most basic way to handle this problem is to use a "callback", and a callback is just a function that is called ("back") once an asynchronous process has completed. You can think of it this way: if you call someone and ask them do something for you that might take some time, you might also ask them to give you a "call back" when they are done so that 1) you know they are done, and 2) they can give you any information that you might need so that you can move forward with your part of the task. That's really all a callback is, and when we write them in our code all we are doing is explicitly organizing this process of coordination as part of the logic of the flow of our application.
+
+Enough theory. Here's how we would use callbacks inside a jQuery AJAX request to perform a search on Google's Books API for works related to peanut butter sandwiches:
+
+```
+var successCallback = function(data) {
+  // Display results on the page.
+};
+
+var errorCallback = function() {
+  alert('Oh no! Something went wrong while fetching results from the server.');
+}
+
+function fetchResults(searchTerm) {
+  $.ajax({
+    method: 'GET',
+    url: 'http://www.googleapis.com/books/v1/volumes?q=' + searchTerm,
+    success: successCallback,
+    error: errorCallback
+  });
+}
+
+fetchResults('peanut butter sandwich');
+```
+
+So let's look at this code. The first two var declarations create two callback functions: `successCallback` and `errorCallback`. Then we define a function `fetchResults` that contains a jQuery AJAX request. The jquery ajax method takes an object with some options set: the first two setup the request itself by specifying the url and the method. Then there are two additional parameters `success` and `error` and here is where we link in our callback functions. Now, once this request returns back either sucessfully or with an error, the appropriate callback function will be called.
+
+The beauty of this is that the (asynchronous) process that occurs on the server, which actually finds and returns the results, can take as long as it needs. Obviously, we don't want it to take a long time since that would annoy the user who is no doubt impatiently waiting for the results, but our program's logic can now handle the indeterminate time-gap of the async process because the process that is fetching the results now knows how to "call back" the process that will display the results once they are in.
+
+So this is what a callback is. Essentially, callbacks are functions that, as one expert has succinctly put it, "wrap or encapsulate the continuation of [a] program" at some unspecified point later.
```

Comment:
@AnnJohn  I think it's good like this, but then I like writing that gives code a "close reading".  (I think it generates focus in the reader effectively).  I'm thinking, though, that that last sentence with the quote from Kyle Simpson could be put up with the other spot where I define callbacks at LN54. Not sure.  I think it also ends nicely by giving a different perspective on what they've already learned in one way.

@pletcher I think a footnote would be appropriate. Do you guys have a preferred way of doing that? I'll work in a mention of `$.get` as an alternative implementation. I agree it's a good idea because it's that way of writing callacks that one sees more frequently.